src/core/lwt.mli

```diff
@@ -21,502 +22,1785 @@
+    [let%lwt]. That is the recommended syntactic sugar for {!Lwt.bind}, and is
+    pronounced “bind”:
+
+{[
+let () =
+  let p : unit Lwt.t =
+    let%lwt line_1 = Lwt_io.(read_line stdin) in
+    let%lwt line_2 = Lwt_io.(read_line stdin) in
+    Lwt_io.printf "%s and %s\n" line_1 line_2
+  in
+
+  Lwt_main.run p
+
+(* ocamlfind opt -linkpkg -package lwt.ppx,lwt.unix code.ml && ./a.out *)
+]}
+
+    The way that works is everything in scope after the “[in]” in
+    “[lwt%lwt x =] ... [in] ...” goes into a callback, and “[x]” is that
+    callback's argument. So, we could have been very explicit, and written the
+    code like this:
+
+{[
+let () =
+  let p : unit Lwt.t =
+    let line_1_promise : string Lwt.t = Lwt_io.(read_line stdin) in
+    Lwt.bind line_1_promise (fun (line_1 : string) ->
+
+      let line_2_promise : string Lwt.t = Lwt_io.(read_line stdin) in
+      Lwt.bind line_2_promise (fun (line_2 : string) ->
+
+        Lwt_io.printf "%s and %s\n" line_1 line_2))
+  in
+
+  Lwt_main.run p
+
+(* ocamlfind opt -linkpkg -package lwt.unix code.ml && ./a.out *)
+]}
+
+    But, as you can see, this is verbose, and the indentation gets a bit crazy.
+    So, we will always use [lwt%lwt].
+
+    The code above reads two lines in sequence, because we waited for [line_1],
+    before calling the second {!Lwt_io.read_line} in the callback, to start the
+    second I/O.
+
+    We could also run I/O {e concurrently}. All we have to do is not start the
+    second I/O in a callback of the first. Because it doesn't make sense to read
+    two lines from STDIN concurrently, let's start two waits instead:
+
+{[
+let () =
+  Lwt_main.run begin
+    let three_seconds : unit Lwt.t = Lwt_unix.sleep 3. in
+    let five_seconds : unit Lwt.t = Lwt_unix.sleep 5. in
+    let%lwt () = three_seconds in
+    let%lwt () = Lwt_io.printl "3 seconds passed" in
+    let%lwt () = five_seconds in
+    Lwt_io.printl "Only 2 more seconds passed"
+  end
+
+(* ocamlfind opt -linkpkg -package lwt.ppx,lwt.unix code.ml && ./a.out *)
+]}
+
+    This program takes about five seconds to run. We are still new to [let%lwt],
+    so let's desugar it:
+
+{[
+let () =
+  Lwt_main.run begin
+    let three_seconds : unit Lwt.t = Lwt_unix.sleep 3. in
+    let five_seconds : unit Lwt.t = Lwt_unix.sleep 5. in
+
+    (* Both waits have already been started at this point! *)
+
+    Lwt.bind three_seconds (fun () ->
+      (* This is 3 seconds later. *)
+      Lwt.bind (Lwt_io.printl "3 seconds passed") (fun () ->
+        Lwt.bind five_seconds (fun () ->
+          (* Only 2 seconds were left in the 5-second wait, so
+              this callback runs 2 seconds after the first callback. *)
+          Lwt_io.printl "Only 2 more seconds passed")))
+  end
+
+(* ocamlfind opt -linkpkg -package lwt.unix code.ml && ./a.out *)
+]}
+
+    And that's it! Concurrency in Lwt is simply a matter of whether you start an
+    operation in the callback of another one or not.
```

Comment:
Maybe also mention that there are other operators (e.g., `join`) which are shown later.